### ISSUE TITLE
Unhandled exception on 32-bit but not 64-bit

### ISSUE THREAD

**Issue by enderandrew** [Aleph-One-Marathon/alephone#306]
Plugin:

http://simplici7y.com/items/marathon-infinity-upscaled-monsters-art

Running that plugin by itself causes "Marathon Infinity.exe" and "Marathon 2.exe" to crash every time when starting a new game. I removed other plugins to make sure it wasn't a conflict with another plugin. However, if I use the 64-bit executable the game runs fine with the plugin.

I'm using the latest build from April 8th, 2021.

The log file shows this error:

Unhandled exception: std::bad_alloc (../../git/Source_Files/shell.cpp:291)

**Comment by aaronfreed** [comment]
Almost certainly an out of memory error – 32-bit Windows apps are limited to 2 GB of RAM, and the plugin is most likely exceeding that (which is surprising if you’re running vanilla M2/∞ maps, since vanilla M2 should never load more than three monster collections per level, and most vanilla M∞ levels should load at most four). There’s really nothing that can be done to fix this (besides allocating memory more efficiently, but that would have very little impact on this plugin specifically). However, the 64-bit build will apparently become the default download for the next release of Aleph One, since almost no one still runs 32-bit Windows, which is nice.

But yeah, the only solution to this is just “use the 64-bit Windows build”. (That’s also the best solution for MaraToon and _Eternal_ 1.2.0 for now, although _Eternal_ 1.2.1 will hopefully be released soon™.)

**Comment by enderandrew** [comment]
I did test one of those 4 GB patch tools to make the 32-bit "Marathon Infinity.exe" executable supposedly support up to 4 GB of RAM, but it still crashes. The zip file for the plugin is 200 MB and extracted it is 800 MB. Is it really using more than 2 or 4 GB of RAM for Marathon with that plugin?

I just fired up the 64-bit version and when I start a new game and fully load in with all my plugins, I show the process using 2.6 GB of RAM. More than the 32-bit client can handle out of the box. Either the 4 GB patch tool isn't working, or it isn't enough to prevent this crash.

But seeing 64-bit as the default moving forward is nice.
